### PR TITLE
added system resource

### PR DESCRIPTION
**Pull request by shenda1**
# Description
PR is raised to add system resource.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 80% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Acceptance Tests

image

### CODE REVIEW

File: powerflex/provider/system_resource.go [hunk 1]
```diff
@@ -0,0 +1,545 @@
...
+
+// SystemResourceSchema defines the schema for system resource
+var SystemResourceSchema schema.Schema = schema.Schema{
+	Description:         "This resource is used to manage the cluster level operations of PowerFlex Array. This resource supports Create, Update and Delete operations.",
+	MarkdownDescription: "This resource is used to manage the cluster level operations of PowerFlex Array. This resource supports Create, Update and Delete operations.",
+	Attributes: map[string]schema.Attribute{
+		"id": schema.StringAttribute{
+			Computed:            true,
+			Description:         "Placeholder",
+			MarkdownDescription: "Placeholder",
```
Comment (rounak-adhikary) [inline powerflex/provider/system_resource.go:100]: Only one comment, the description here needs update.

File: powerflex/provider/system_resource.go [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+	SdcGuids := make([]string, 0)
+	diags.Append(plan.SdcGuids.ElementsAs(ctx, &SdcGuids, true)...)
+
+	if plan.RestrictedMode.ValueString() == "Guid" {
+		for _, SdcGUID := range SdcGuids {
+			diags.Append(r.ApproveSdcGUID(SdcGUID)...)
+		}
+	}
+
+	if !plan.SdcApprovedIPs.IsNull() {
```
Comment (AnikaAgiwal2711) [inline powerflex/provider/system_resource.go:227]: Can we add comments so that it becomes more readable.

**Review by rounak-adhikary — DISMISSED** [review]

**Comment by shenda1** [comment]
run e2e test

**Review by AnikaAgiwal2711 — DISMISSED** [review]

**Comment by shenda1** [comment]
run e2e test

**Comment by shenda1** [comment]
run e2e test

**Comment by shenda1** [comment]
run e2e test

**Review by rounak-adhikary — APPROVED** [review]

**Review by AnikaAgiwal2711 — APPROVED** [review]

**Pull request merged**